src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/Frame.FeatureCollection.cs

```diff
@@ -202,6 +203,28 @@ string IHttpRequestIdentifierFeature.TraceIdentifier
             set => TraceIdentifier = value;
         }
 
+        long? IHttpMaxRequestBodySizeFeature.MaxRequestBodySize
+        {
+            get => MaxRequestBodySize;
+            set
+            {
+                if (HasStartedConsumingRequestBody)
```

Comment:
Yes, the problem is you can add middleware anywhere in the pipeline that reads the body. There needs to be a way to not throw here.

> Do you want to add HasStartedConsumingRequestBody to some request feature? Do you want to do the same for WasUpgraded?

Yes, or something more generally named like our other bools of this nature (HasResponseStarted etc).